File: backend/examply_backend/dev.py
"""
Development server startup script with automatic environment setup.
"""
import os
import shutil
import sys
from pathlib import Path

def setup_env_file():
    """Setup .env file if it doesn't exist."""
    backend_dir = Path(__file__).parent.parent
    env_file = backend_dir / ".env"
    env_example = backend_dir / ".env.example"

    if not env_file.exists() and env_example.exists():
        shutil.copy(env_example, env_file)
        print(f"✅ Created {env_file}")
    elif not env_file.exists():
        # Create minimal .env file
        with open(env_file, "w") as f:
            f.write("# Backend Environment Variables\n")
            f.write("DATABASE_URL=sqlite:///./examply.db\n")
            f.write("API_HOST=0.0.0.0\n")
            f.write("API_PORT=8000\n")
            f.write("DEBUG=true\n")
        print(f"✅ Created minimal {env_file}")

def main():
    """Main development server startup."""
    print("🚀 Starting Examply backend development server...")

    # Setup environment file
    setup_env_file()

    # Start uvicorn server
    try:
        import uvicorn
        uvicorn.run(
            "app.main:app",
            host="0.0.0.0",
            port=8000,
            reload=True,
            reload_dirs=["app"]
        )
    except ImportError:
        print("❌ uvicorn not found. Run: uv sync")
        sys.exit(1)
    except KeyboardInterrupt:
        print("\n👋 Backend server stopped")

if __name__ == "__main__":
    main()

File: frontend/scripts/dev.js
/**
 * Development server startup script with automatic environment setup.
 */
const fs = require('fs');
const path = require('path');
const { spawn } = require('child_process');

function setupEnvFile() {
  const envFile = path.join(__dirname, '..', '.env.local');
  const envExample = path.join(__dirname, '..', '.env.local.example');

  if (!fs.existsSync(envFile)) {
    if (fs.existsSync(envExample)) {
      fs.copyFileSync(envExample, envFile);
      console.log(`✅ Created ${envFile}`);
    } else {
      // Create minimal .env.local file
      const envContent = `# Frontend Environment Variables
NEXT_PUBLIC_API_URL=http://localhost:8010
NEXT_PUBLIC_ENVIRONMENT=development
`;
      fs.writeFileSync(envFile, envContent);
      console.log(`✅ Created minimal ${envFile}`);
    }
  }
}

function main() {
  console.log('🚀 Starting Examply frontend development server...');

  // Setup environment file
  setupEnvFile();

  // Start Next.js development server
  const nextProcess = spawn('npx', ['next', 'dev', '--turbopack'], {
    stdio: 'inherit',
    shell: true
  });

  // Handle process termination
  process.on('SIGINT', () => {
    console.log('\n👋 Frontend server stopped');
    nextProcess.kill('SIGINT');
    process.exit(0);
  });

  nextProcess.on('error', (error) => {
    console.error('❌ Failed to start frontend server:', error);
    process.exit(1);
  });

  nextProcess.on('close', (code) => {
    if (code !== 0) {
      console.error(`❌ Frontend server exited with code ${code}`);
      process.exit(code);
    }
  });
}

main();